### PR TITLE
Fix SimpleClient Broker API version default

### LINKED ISSUE #902: BrokerConnection API_VERSION has wrong default value in docstring
I'm confused about the API_VERSION default value.

[`conn.py`](https://github.com/dpkp/kafka-python/blob/master/kafka/conn.py#L77) says:
> 'api_version': (0, 8, 2), # default to most restrictive

But the [docstring](https://github.com/dpkp/kafka-python/blob/master/kafka/conn.py#L139) for that class says:

>  api_version (tuple): specify which kafka API version to use. Accepted
                values are: (0, 8, 0), (0, 8, 1), (0, 8, 2), (0, 9), (0, 10)
                If None, KafkaClient will attempt to infer the broker
                version by probing various APIs. *Default: None*

So the default isn't `None`, it's `(0, 8, 2)`. Which should be fixed, the docstring or the default value?

I'm *think* this class is only used by SimpleClient not by the newer KafkaClient async, since I see another set of values here:
https://github.com/dpkp/kafka-python/blob/master/kafka/client_async.py#L78 where both the code and the docstring state the default value is `None`.

Help?

### PR DESCRIPTION
**Pull request by jeffwidman**
Fix #902

### CODE REVIEW

File: kafka/conn.py [hunk 1]
```diff
@@ -135,8 +135,8 @@ def __init__(self, host, port, afi, **configs):
                 default: None.
             api_version (tuple): specify which kafka API version to use. Accepted
                 values are: (0, 8, 0), (0, 8, 1), (0, 8, 2), (0, 9), (0, 10)
-                If None, KafkaClient will attempt to infer the broker
-                version by probing various APIs. Default: None
+                If None, SimpleClient will attempt to infer the broker
```
Comment (dpkp) [inline kafka/conn.py]: Both the new KafkaClient and the old SimpleClient use BrokerConnection, so we should probably describe just the BrokerConnection behavior here

**Comment by jeffwidman** [comment]
Updated, also cleaned up docstrings while I was in there to be more consistent across the various clients.

**Comment by jeffwidman** [comment]
I just rebased to fix merge conflicts...

**Comment by jeffwidman** [comment]
Rebased to fix merge conflicts.

**Comment by jeffwidman** [comment]
Rebased

**Comment by jeffwidman** [comment]
Anything need changing on this one?

**Comment by dpkp** [comment]
Thanks!

**Comment by jeffwidman** [comment]
Thanks for merging this one

**Pull request merged**